### ISSUE TITLE
Prepare integration tests to be executable as independent binary

### ISSUE THREAD

**Issue by agajdosi** [minishift/minishift#2181]
For testing purposes of centos.iso repository and similar cases it would be useful to have integration test accessible as independent binary. Thus enabling running integration tests on machines without Golang installed. By running `go test github.com/minishift/minishift/test/integration --tags=integration -c` in root of Minishift repository we get `integration.test` binary in the same place.

This binary can be relatively nicely used:
```
./integration.test --binary=/home/agajdosi/minishift/minishift --paths=test/integration/features --tags=cmd-addons
```
and because we use flag package to get arguments between Makefile and itests, help is quite useful and nice also:
```
[agajdosi@dhcp-24-156 minishift]$ ./integration.test -h
Usage of ./integration.test:
  -binary string
        Path to minishift binary.
  -copy-oc-from string
        Path to binary, the binary will $MINISHIFT_HOME/cache/oc.
  -definitions

  -format string
        Sets which format godog will use. (default "pretty")
  -minishift-args string
        Arguments to pass to minishift.
  -no-colors
        Disable colors in godog output.
  -paths string
         (default "./features")
  -run-before-feature string
        Set of minishift commands to be executed before every feature. Individual commands must be delimited by a semicolon.
  -stop-on-failure
        Stop when failure is found.
  -tags string
        Tags for godog test.
  -test-dir string
        Path to the directory in which to execute the tests. (default "./out/integration-test")
  -test-with-specified-shell string
        Name of shell to be used for steps which executes commands directly in persistent shell instance.
  -test.bench regexp
        run only benchmarks matching regexp
  -test.benchmem
        print memory allocations for benchmarks
[...]
```
It however has some problems, mostly path definitions and similar stuff, which needs some care. Once it is implemented we might release the integration.test binaries in different repository to make it available for CI and other teams.

**Comment by agajdosi** [comment]
Path problems will be resolved with #2339.

**Comment by LalatenduMohanty** [comment]
Now that #2339 fixed, we can fix this issue.
